src/lib/common.ts

```diff
@@ -25,17 +25,13 @@ export function setValueOutgoingMessage(
   return { result };
 }
 
-export type ControllerFirmwareUpdateResultType =
-  | { result: ControllerFirmwareUpdateResult } // schemaVersion >= 29
-  | { success: boolean }; // schemaVersion < 29
-
-export type NodeFirmwareUpdateResultType =
-  | { result: FirmwareUpdateResult } // schemaVersion >= 29
+export type FirmwareUpdateResultType =
+  | { result: ControllerFirmwareUpdateResult | FirmwareUpdateResult } // schemaVersion >= 29
```

Comment:
I don't know what is usually done here, but this type means that any client for this server will need to go check what the driver actually returns for each command that has this return type (that's basically already done on documentation anyway, but it makes it harder if you are checking an older server version, as the docs aren't versioned).

Reply:
yeah this isn't the best way to type this, I've been playing with other options but I haven't found a good alternative yet. With that being said, since all of the communication is happening over websocket, the typing is really for typechecking and code quality for the server